resources/run_adl.sh

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+NAME=`basename "$0"`
+if [[ $SKIP_RUN_ADL -eq 1 ]]; then
+  echo "Skipping $NAME"
+  exit 0
+fi
+
+ARG=`echo $1 | tr '[:upper:]' '[:lower:]'`
+if [[ -z $ARG ]]; then
+  echo "Usage: $NAME <operation>"
+  echo "operation: 'start' or 'stop'"
+  exit 0
+fi
+
+GO_VERSION="go1.16"
+if [ -d "/opt/golang/$GO_VERSION" ]; then
+  GOROOT="/opt/golang/$GO_VERSION"
+  GOBINDIR="$GOROOT"/bin
+  PATH=$GOBINDIR:$PATH
+fi
+
+TMP_DIR="/tmp/run_adl/"
+MONGOHOUSE_URI=git@github.com:10gen/mongohouse.git
+MONGOHOUSE_DIR=$(pwd)/mongohouse
+MONGO_DB_PATH=$(pwd)/test_db
+LOGS_PATH=$(pwd)/logs
+MONGOD_PORT=28017
+MONGOHOUSED_PORT=27017
+START="start"
+STOP="stop"
+MONGOD="mongod"
+MONGOHOUSED="mongohoused"
+MONGO_DOWNLOAD_LINK=
+OS=$(uname)
+
+MONGO_DOWNLOAD_BASE=https://fastdl.mongodb.org
+# Ubuntu 18.04
+MONGO_DOWNLOAD_UBUNTU=mongodb-linux-x86_64-ubuntu1804-5.0.4.tgz
+# RedHat 7
+MONGO_DOWNLOAD_REDHAT=mongodb-linux-x86_64-rhel70-5.0.4.tgz
+# macOS
+MONGO_DOWNLOAD_MAC=mongodb-macos-x86_64-5.0.4.tgz
+
+check_procname() {
+  ps -ef 2>/dev/null | grep $1 | grep -v grep >/dev/null 
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_port() {
+  netstat -van 2>/dev/null | grep LISTEN | grep $1 >/dev/null
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_mongod() {
+  check_procname $MONGOD
+  process_check_result=$?
+  check_port $MONGOD_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+check_mongohoused() {
+  check_procname $MONGOHOUSED
+  process_check_result=$?
+  check_port $MONGOHOUSED_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# check if mac or linux
+if [ $OS = "Linux" ]; then
+  distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+  if [ "$distro" = "\"Red Hat Enterprise Linux\"" ] ||
+[ "$distro" = "\"Red Hat Enterprise Linux Server\"" ]; then
+    export VARIANT=rhel7
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_REDHAT
+  elif [ "$distro" = "\"Ubuntu\"" ]; then
+    export VARIANT=ubuntu1804  
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_UBUNTU
+  else
+    echo ${distro} not supported
+    exit 1
+  fi
+elif [ $OS = "Darwin" ]; then
+  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_MAC
+else
+  echo $(uname) not supported
+  exit 1
+fi
+
+check_mongod
+if [[ $? -ne 0 ]]; then
+  if [ $ARG = $START ]; then
+    # Install and start mongod
```

Comment:
Let's consider adding `echo "Starting mongod"` like we have logging for stopping it in the else branch.